### ISSUE TITLE
How refit on train+val set?

### ISSUE THREAD

**Issue by EdgedSquirrels** [Nixtla/neuralforecast#1124]
How should I refit the model on the whole dataset containing both training and validation sets?

It seems that both `nf.fit` and `nf.cross_validation` don't support this feature, but AutoModels have `refit_with_val`. Could I set `refit_with_val` on non-auto models?

**Comment by marcopeix** [comment]
You can simply combine both sets into a single dataframe and pass it to the `fit` method. Alternatively, if you used `val_size`, simply set it to 0 to use all of the data. 

With `cross_validation`, you can use the `refit` parameter and it to `True`.

**Comment by EdgedSquirrels** [comment]
Hi @marcopeix,

Thanks for your answer. I’m not sure that when early stopping is enabled, will the number of training steps recorded and used during `refit`, or it will use `max_steps` during `refit`?

Also I’ve tried `cross_validation` with `refit=True`, but I only saw a progress bar running without another progress bar running after that. I’m not sure that whether `cross_validation` supports `refit` when `val_size` is given instead of `n_windows`.

**Comment by andmadeira** [comment]
Hi @marcopeix !

Taking @EdgedSquirrels's  question about early stopping with `refit`, I noticed that once the early stopping is triggered, the fit for that window stops, and when moving to the following refit window, the model is fitted at least until the first validation step, where the refit is imemdiately stopped in case the validation loss is above the previous window's lowest, not considering patience.
I understand that a refit will continue to fit the model from where it last stopped. Shouldn't the patience counter be reset on every refit?
